Data/Vector/Generic.hs

```diff
@@ -1381,8 +1381,19 @@ takeWhile f = unstream . Bundle.takeWhile f . stream
 -- | /O(n)/ Drop the longest prefix of elements that satisfy the predicate
 -- without copying.
 dropWhile :: Vector v a => (a -> Bool) -> v a -> v a
-{-# INLINE dropWhile #-}
-dropWhile f = unstream . Bundle.dropWhile f . stream
+{-# INLINE_FUSED dropWhile #-}
+-- In the case that the argument is an actual vector,
+-- this is a faster solution than stream fusion.
+dropWhile f xs = case findIndex (not . f) xs of
+                   Just i  -> unsafeDrop i xs
+                   Nothing -> empty
+
+-- If the argument to 'dropWhile' comes from a stream,
+-- we need to avoid unnecessary allocation.
```

Comment:
I think it's better to say that we want to avoid creating a new vector. Comment doesn't explain what is being allocated. 

Also please add mention that  `new (New.unstream ` is body of unstream.